### DISCUSSION TITLE
Why must `docker.sock` be mounted inside the MegaLinter container?

### DISCUSSION THREAD

**Discussion by sanmai-NL** [oxsecurity/megalinter#2722] [Ideas]
There are multiple mechanisms to schedule containers on a target host. Rather than instructing users to mount this socket by default, which introduces a security weakness, I propose that we offer options.

- Describe `$DOCKER_HOST` and options to set it up.
- Offer Kubernetes manifests.
- Offer means to enhance existing Containerfiles to add linters and MegaLinter tooling, rather than building a separate MegaLinter image.
- Describe ways to integrate with popular CI systems and their container engines.

**Comment by nvuillam** [comment]
Few linters (maybe swiftlint only) are run using directly their docker image, that's what docker.sock must be sent.... except if you don't use swiftlint :)

**Reply by sanmai-NL** [reply]
This violates [rule #1 of the OWASP Docker Security checklist](https://cheatsheetseries.owasp.org/cheatsheets/Docker_Security_Cheat_Sheet.html#rule-1-do-not-expose-the-docker-daemon-socket-even-to-the-containers). Seems like a huge price to pay to support Swiftlint. And I proposed some alternative mechanisms in the opening post.

**Reply by nvuillam** [reply]
I'm not sure swiftlint is very used, maybe we could indeed deprecate it in a future version if we don't find a way to install it directly on our docker image

I'm no k8 expert but if you want to make POC, they are very welcome :) ( cc @Kurt-von-Laven @bdovaz @echoix )

**Reply by sanmai-NL** [reply]
Writing a few Kubernetes manifests is easy, but the architecture would be different. Instead of writing large Containerfiles, you would define a set of preconfigured upstream container images. See https://github.com/oxsecurity/megalinter/discussions/2719
